<robot name="f1">
  <link name="base" >
   <visual>
       <geometry>
        <box size=".195 .145 .04"/>
       </geometry>
       <origin xyz="0 0 .02" rpy="0 0 0" />
       <material name="blue">
        <color rgba="0 0 .8 .8"/>
       </material>	
    </visual>
  </link>
  <link name="link1" >
   <visual>
       <geometry>
         <box size=".08 .06 .035"/>
       </geometry>
       <origin xyz="0 0 .0175" rpy="0 0 0" />
       <material name="red">
        <color rgba=".8 0 0 .8"/>
       </material>	
    </visual>
  </link>
  <link name="link2" >
   <visual>
       <geometry>
         <cylinder length=".13" radius=".005"/>
       </geometry>
       <origin xyz="0 0 .065" rpy="0 0 0" />
       <material name="grey_a">
        <color rgba=".4 .6 .4 .8"/>
       </material>	
    </visual>
  </link>
  <link name="link3" >
   <visual>
       <geometry>
         <box size=".065 .03 .06"/>
       </geometry>
       <origin xyz="0 0 .03" rpy="0 0 0" />
       <material name="white">
        <color rgba="1 1 1 .8"/>
       </material>	
    </visual>
  </link>
  <link name="link4" >
   <visual>
       <geometry>
         <cylinder length=".11" radius=".005"/>
       </geometry>
       <origin xyz="0 0 .055" rpy="0 0 0" />
       <material name="green">
        <color rgba="0 .8 0 1"/>
       </material>	
    </visual>
  </link>
  <link name="polso" >
   <visual>
       <geometry>
         <cylinder length=".005" radius=".005"/>
       </geometry>
       <origin xyz="0 0 .0025" rpy="0 0 0" />
       <material name="white_punta">
        <color rgba="1 1 1 .5"/>
       </material>	
    </visual>
  </link>
  <link name="punta" >
   <visual>
       <geometry>
         <cylinder length=".005" radius=".005"/>
       </geometry>
       <origin xyz="0 0 .0025" rpy="0 0 0" />
       <material name="red_punta">
        <color rgba="1 0 0 1"/>
       </material>	
    </visual>
  </link>

  <joint name="joint1" type="revolute">
    <parent link="base"/>
    <child link="link1"/>
    <origin xyz=".0375 0 .045" rpy="0 0 0" />
    <axis xyz="0 0 1"/>
    <limit effort="30" velocity="1.0" lower="-1.57" upper="1.57" />
  </joint>

  <joint name="joint2" type="revolute">
    <parent link="link1"/>
    <child link="link2"/>
    <origin xyz="0 0 .04" rpy="0 0 0" />
    <axis xyz="1 0 0" />
    <limit effort="30" velocity="1.0" lower="-1.57" upper="1.57" />
  </joint>

  <joint name="joint3" type="revolute">
    <parent link="link2"/>
    <child link="link3"/>
    <origin xyz="0 0 .135" rpy="0 0 0" />
    <axis xyz="1 0 0" />
    <limit effort="30" velocity="1.0" lower="-0.4" upper="1.57" />
  </joint>

  <joint name="joint4" type="revolute">
    <parent link="link3"/>
    <child link="link4"/>
    <origin xyz="0 0 .065" rpy="0 0 0" />
    <axis xyz="0 0 1" />
    <limit effort="30" velocity="1.0" lower="-1.57" upper="1.57" />
  </joint>

  <joint name="rigido" type="fixed">
    <parent link="link4"/>
    <child link="polso"/>
    <origin xyz="0 0 .115" rpy="0 0 0" />
  </joint>

  <joint name="joint_strumento" type="fixed">
    <parent link="polso"/>
    <child link="punta"/>
    <origin xyz="0 0 .01" rpy="0 0 0" />
  </joint>

  <transmission name="tran1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint1"> 
	<hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="joint_1">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name="tran2">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint2">
	<hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="joint_2">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name="tran3">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint3">
	<hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="joint_3">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name="tran4">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint4">
	<hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="joint_4">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

</robot>
